Assets/Scripts/Singleton.cs

```diff
@@ -11,20 +11,22 @@ public static T Instance
     {
         get
         {
-            if (instance == null)
-            {
-                instance = FindObjectOfType<T>();
-
-                DontDestroyOnLoad(instance);
-            }
-            else if(instance != FindObjectOfType<T>())
-            {
-                Destroy(instance);
-            }
-
             return instance;
         }
     }
 
+    private void Awake()
+    {
+        if(instance == null)
+        {
+            instance = this as T;
+            DontDestroyOnLoad(this.gameObject);
+        }
+        else if(instance != null)
+        {
+            Destroy(this.gameObject);
```

Comment:
don't destroy the entire GO. just destroy this component - coz there might be other components on the same GO